docs/customize/dois.md

```diff
@@ -84,3 +84,22 @@ DATACITE_DATACENTER_SYMBOL = "CERN.INVENIORDM"
   records, thus metadata like titles, authors, description and more is sent to
   DataCite Metadata registry where it is public. Provide an external DOI if
   no DOI should be registered by InvenioRDM.
+
+
+# External DOIs
+
+In some instances, the generation of DOIs is not necessary or even wanted.
```

Comment:
Hello! If the point is to use only external identifier, is it worth to check if the value can be inputted in the alternative identifiers instead? ping @tmorrell

Reply:
I am not @tmorrell but this is something, but I ran into the weirdnesses surrounding DOIs in our instance too, that I may have valuable insight.

Using only the alternative identifier is typically not enough because it doesn't get into the citation and doesn't get some of the checks/assurances that "external" provides.

The settings described in this PR only "work" (:wink: see next paragraph) if your instance is ONLY an aggregator. If, like ours actually, you are both an aggregator of other works AND a minter of new ones, you can't use those settings.  

More importantly: quotation mark around _work_ here because these settings cause the DataCiteClient to have invalid credentials and therefore fail in its asynchronous task trying to register a DOI (see https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/pids/providers/datacite.py#L53-L77). In other words, this is telling people to put their instance in an invalid state, so that a background process, DOI registration, fails (without any visible sign to the instance user). This is shaky grounds.

Reply:
Yes, I felt it -- I mean, the unstable ground --, @fenekku when I was down there using `None` on `DATACITE_*` settings... But, as you noticed, my instance is indeed _only_ an aggregator, and after making it "work" with the help from Discord (linked at the top), somebody suggested this PR.

Sure enough, if those settings put Invenio in an invalid state, then we should close this (doc) PR and, actually, open a (bug) issue for that (`None`) could not happen.

Before I go, though: what if I set `DATACITE_TEST_MODE = True`? I understand this would disable interactions to/from Datacite; Would that cause the "invalid state" to run unnoticed? (so I could sleep in peace...)

Reply:
Setting `DATACITE_TEST_MODE = True` would cause the test DataCite servers to be used: https://github.com/inveniosoftware/datacite/blob/master/datacite/rest_client.py#L51 . So kind of same situation :laughing: 

Let's be constructive though! I think what you want is perhaps achievable by configuring these: https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/config.py#L272-L326 

There might be a way, starting from there to figure out how you should craft your DataCitePIDProvider-like to achieve what you want. 

It may also be a case of submitting a ticket for this feature ;)